src/xtgeo/interfaces/rms/_rms_base.py

```diff
@@ -0,0 +1,86 @@
+"""Shared base for RMS RegularSurface reader/writer."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from xtgeo.common.log import null_logger
+from xtgeo.interfaces.rms.rmsapi_utils import (
+    RmsApiUtils,
+    _StorageTypeRegularSurface as StorageType,
+)
+
+if TYPE_CHECKING:
+    from xtgeo.interfaces.rms._rmsapi_package import RmsProjectType
+
+logger = null_logger(__name__)
+
+__all__ = ["_BaseRmsDataRW"]
+
+
+class _BaseRmsDataRW:
+    """
+    Common init, input validation etc for read/write RMS RegularSurface, Points, etc.
+    """
+
+    def __init__(
+        self,
+        project: RmsProjectType,
+        name: str,
+        category: str,
+        stype: str,
+        realisation: int = 0,
+    ):
+        self.project = project
+        self.name = name
+        self.category = category
+        self.stype = stype
+        self.realisation = realisation
+
+        # Internal state
+        self._rmsapi_utils: RmsApiUtils | None = None
+        self._stype_enum: StorageType | None = None
+
+    def _init_utils(self, readonly: bool) -> None:
+        """Create RmsApiUtils with the selected readonly mode."""
+        self._rmsapi_utils = RmsApiUtils(self.project, readonly=readonly)
```

Comment:
```suggestion
    def get_rms_utils(self, readonly: bool = False) -> RmsApiUtils:
        """Create RmsApiUtils with the selected readonly mode."""
        if self._rmsapi_utils is None:
            self._rmsapi_utils = RmsApiUtils(self.project, readonly=readonly)
        return self._rmsapi_utils
```

I think doing a getter like this will make the typing simpler. Then every caller does not need to worry about whether it's `None` or not. It'd be even simpler if `readonly` was not an argument, and you could just make it a `@property` that initializes the inner value when it's needed.